File: ResourceManager/Services/Abstractions/ITagDataService.cs
using ResourceManager.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ResourceManager.Services.Abstractions
{
    public interface ITagDataService : IDataService<Tag>
    {
        Task<IList<Tag>> GetTagObjectForPasswordID(int id);
        Task<IList<Tag>> GetTagObjectForBookID(int id);
        Task<IList<Tag>> GetTagObjectForArticleID(int id);
        Task<IList<Tag>> GetTagObjectForRepositoryID(int id);
        Task<IList<Tag>> GetTagObjectForActivityID(int id);
        Task<IList<Tag>> GetTagObjectForNoteID(int id);
    }
}

File: ResourceManager/Services/Abstractions/ITagDataService.vb
Imports ResourceManager.Models
Imports System.Collections.Generic
Imports System.Threading.Tasks

Namespace ResourceManager.Services.Abstractions
    Public Interface ITagDataService
        Inherits IDataService(Of Tag)
        Function GetTagObjectForPasswordID(ByVal id As Integer) As Task(Of IList(Of Tag))
        Function GetTagObjectForBookID(ByVal id As Integer) As Task(Of IList(Of Tag))
        Function GetTagObjectForArticleID(ByVal id As Integer) As Task(Of IList(Of Tag))
        Function GetTagObjectForRepositoryID(ByVal id As Integer) As Task(Of IList(Of Tag))
        Function GetTagObjectForActivityID(ByVal id As Integer) As Task(Of IList(Of Tag))
        Function GetTagObjectForNoteID(ByVal id As Integer) As Task(Of IList(Of Tag))
    End Interface
End Namespace
